README.md

```diff
@@ -0,0 +1,103 @@
+# Hazelcast Open Binary Client Protocol
+
+**Hazelcast Open Binary Client Protocol** definitions and code generator for multiple programming languages.
+
+## Hazelcast Open Binary Client Protocol Definitions
+
+The protocol is defined in `protocol-definitions/*.yaml` yaml files where each yaml file represents a service like Map, List, Set etc.
+
+## Service definition
+
+A service is defined by a separate YAML file, containing all its method definitions.
+
+```yaml
+id: Service Id (1-255)
+name: Service Name
+methods:
+  - id: METHOD-ID-1 (1-255)
+    name: METHOD-NAME-1
+    ...
+  - id: METHOD-ID-2 (1-255)
+    name: METHOD-NAME-2
+    ...
+```
+
+A method(aka Remote method call) is defined by a request-response pair.
+
+A basic method structure example:
+
+```yaml
+  - id: METHOD-ID-1 (1-255)
+    name: METHOD-NAME-1
+    since: 2.0
+    doc: |
+       Documentation of the method call
+    request:
+      retryable: false
+      acquires-resource: false
+      partition-identifier: None
+      params:
+        - name: parameter1
+          type: String
+          nullable: false
+          since: 2.0
+          doc: |
+             Documentation of the parameter 1
+        - name: parameter1
+          type: Data
+          nullable: false
+          since: 2.0
+          doc: |
+             Documentation of the parameter 2
+    response:
+      params:
+        - name: response parameter 1
+          type: Data
+          nullable: true
+          since: 2.0
+          doc: |
+             the response parameter 1
+
+```
+
+## Code Generator
+
+The new protocol generator generates the related lang codec into the configured folder. It does not depend on hazelcast repo.
+
+
+### Setup
+
+You need the have python3 configured on your `PATH`. After cloning the repository, install the python library dependencies ones:
+
+```bash
```

Comment:
on some users, pip3 (pip backed by python 3) may be named as just `pip` I think we should just mention that this generator requires python 3 and change this to `pip install ...`